ipatests/pytest_ipa/integration/tasks.py

```diff
@@ -795,15 +795,24 @@ def remove_trust_info_from_ad(master, ad_domain, ad_hostname):
     kinit_as_user(master,
                   'Administrator@{}'.format(ad_domain.upper()),
                   master.config.ad_admin_password)
+    # Find cache for the user
+    cache_args = []
+    result = master.run_command(["klist"]).stdout_text
+    pattern = re.compile(r'Ticket cache: (?P<cache>.*)\n')
+    res = pattern.search(result)
+    if res:
+        cache_args = ["--use-krb5-ccache", res['cache']]
+
```

Comment:
Can you break out "Find cache for the user" into a separate task and use it in both places?